pkg/irreparabledb/irreparabledb.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparabledb
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/internal/migrate"
+	dbx "storj.io/storj/pkg/irreparabledb/dbx"
+)
+
+// Database implements the irreparable RPC service
+type Database struct {
+	db *dbx.DB
+}
+
+// RemoteSegmentInfo is info about a single entry stored in the irreparable db
+type RemoteSegmentInfo struct {
+	SegmentPath        []byte
+	SegmentDetail      []byte
+	LostPiecesCount    int64
+	RepairUnixSec      int64
+	RepairAttemptCount int64
+}
+
+// New creates instance of Server
+func New(source string) (*Database, error) {
+	db, err := dbx.Open("sqlite3", source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrate.Create("irreparabledb", db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		db: db,
+	}, nil
+}
+
+// IncrementRepairAttempts a db entry for to increment the repair attempts field
+func (db *Database) IncrementRepairAttempts(ctx context.Context, segmentPath []byte) (err error) {
+	_, err = db.db.Exec(`UPDATE irreparabledbs SET repair_attempt_count = repair_attempt_count + 1 WHERE segmentpath = ?`, segmentPath)
```

Comment:
Does this work both with sqlite and postgres?

Reply:
yep. the db.Exec() will take care of the query based on the open() call to db. (I will verify to be double sure)

https://sqlite.org/lang_UPSERT.html per this website the query format used runs on both Postgres and sql.